src/controllers/tourController.js

```diff
@@ -36,15 +36,33 @@ exports.fetchSingleTour = (req, res) => {
 }
 
 // book a tour
-exports.store = async (req, res) => {
-	// TODO: get user id from url
-	// instantiating new booking
-	const newBooking = new Booking({...req.body});
-	// saving new booking
-	const bookedTour = await newBooking.save();
-	console.log(bookedTour);
-	// send confirmation email
-	await sendConfirmationEmail(bookedTour, req, res);
+exports.bookTour = async (req, res) => {
+	try {
+		// get user id from url
+		let loggedInUser = await User.findById(req.params.id);
+
+		// search for existing tour by id, if no tour exists, create one with logged in user info.
+		Tour.findById(req.body.tour, (err, tour) => {
+			if (err) {
+				return res.status(404).json({message: 'Tour not found'});
+			} else {
+				Booking.create({
+					user: loggedInUser,
+					tour: tour,
+					category: req.body.category,
+					travelType: req.body.travelType
+				}, (err, newBooking) => {
```

Comment:
this booking needs to be pushed into the user's bookedTours list and the user updated data saved